Do-It/Model/Persistence/CoursePersistenceManager.swift

```diff
@@ -9,33 +9,54 @@
 import Foundation
 
 final class CoursePersistenceManager {
+    var encoder = JSONEncoder()
+    var decoder = JSONDecoder()
+    var fileManager = FileManager.default
+    var docsURL: URL
+    var courseURL: URL
     var courses: [Course] {
         didSet {
             saveCoursesToDisk()
         }
     }
 
     init() {
+        do {
+            docsURL = try fileManager.url(for: .documentDirectory, in: .userDomainMask,
+                                          appropriateFor: nil, create: true)
+        } catch {
+            fatalError("The app docs directory wil always exist")
+        }
+        courseURL = docsURL.appendingPathComponent("Courses.json")
         courses = CoursePersistenceManager.loadCoursesFromDisk()
     }
 
-    func save(_ course: Course) {
-
-    }
-
-    func update(_ doIt: DoIt) {
-
-    }
-
-    func delete(_ doIt: DoIt) {
-
+    private static func loadCoursesFromDisk() -> [Course] {
```

Comment:
In hindsight, it makes more sense for this to be an instance method rather than a static method. By making it an instance method, we can remove duplication of `docsURL` and `fileManager`.
You can remove the return type and assign to `courses` directly in the body, and call the method as the last line of the initializer.